checkstyle.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
+)
+
+type checkstyleOutput struct {
+	XMLName xml.Name          `xml:"checkstyle"`
+	Version string            `xml:"version,attr"`
+	Files   []*checkstyleFile `xml:"file"`
+}
+
+type checkstyleFile struct {
+	Name   string             `xml:"name,attr"`
+	Errors []*checkstyleError `xml:"error"`
+}
+
+type checkstyleError struct {
+	Column   int    `xml:"column,attr"`
+	Line     int    `xml:"line,attr"`
+	Message  string `xml:"message,attr"`
+	Severity string `xml:"severity,attr"`
+	Source   string `xml:"source,attr"`
+}
+
+func outputToCheckstyle(issues chan *Issue) int {
+	var lastFile *checkstyleFile
+	out := checkstyleOutput{
+		Version: "5.0",
+	}
+	status := 0
+	for issue := range issues {
+		if lastFile != nil && lastFile.Name != issue.Path {
```

Comment:
Yes. We can reuse the ordering code you have already implemented, thats why I force issues to be ordered by file name in main (main.go:347)

```go
	// Force sorting by path if checkstyle output is selected
	if *outputFlag == "checkstyle" {
		*sortFlag = []string{"path"}
	}
```

This means that if `--output=checkstyle` flag is provided `--sort` flag provided by the user will be ignored. But thats OK because we cannot order issues by anything else in the checkstyle output anyway.